pitch.md

```diff
@@ -1,69 +1,172 @@
+- It should be possible to implement natural deduction, actually!
+- Arguably, it should be impossible to do compile-time calculations that may not terminate, such as Idris requires, but I will not think about that.
+
 
 ### Related work
-_Are there any other DSLs in this domain? If not, describe how you know there
-aren't and conjecture why not. If so, describe them and provide links. How well 
-do they address the need? Are there any particularly admirable qualities of the
-language? Are there parts of the language you think could be improved?_
 
+C++ has templates, which have evolved tremendously and unfortunately.
+D has static arguments, and a lot of control of compile time stuff in some ways.
+Idris has dependent types, which are the sort of thing you would want to
+build in this language. Those are sort of motivating examples.
+
+Closer to what I'm actually going for are Scala and Lisp's macros,
+which allow you to manipulate abstract syntax trees directly,
+give errors based on them, etc. In fact, it may be that if Scala's
+macros are powerful and expressive enough, that is how this project would be implemented.
 
 ## The Project
 This section examines whether the idea makes for a good CS 111 project.
 
 
 ### Suitability
-_What percentage of your time do you think will be spent directly engaging in
-the **language** aspects of this project (e.g., making language design
-decisions), as opposed to "systems" aspects of the project (e.g., implementing a
-complicated semantics that doesn't require a lot of language design)?_
 
+In an attempt to create an actual, practically usable programming language with these features,
+it would take a huge amount of work just on the implementation side.
+But, that is not what I'm interested in. I am only interested in the aspects
+that are associated with compile-time checking and such. On the next level,
+I am also not interested with actually implementing a good type-checking system
+(i.e., one with good inference). The only important thing is that such
+a thing _could_ be made. Furthermore, how well it actually runs is far less important than what
+happens during compilation, _so much so_ that it would be fine if no actually runnable code
+could be made.
+
+Thus, the meat of the project is still in language design. The main question is how to
+give direct access to compile-time calculations. There are basically two ways
+to do this, and it might be possible to unify them. The first is directly manipulating
+abstract syntax trees. The second is a generic form of "property propagation", such
+as allows naive type-checking. That can be described as an operation on ASTs, though.
+So given a rudimentary underlying abstract syntax, which would probably not be any more
+complicated than a very simple subset of Haskell, the main question of the project
+becomes how to expressively manipulate and calculate the properties of ASTs.
 
```

Comment:
Something this has been making me think of is whether or not you could write something like GCC with lots of flags and stuff? Obviously writing a compiler would be really hard but C might let you do almost anything if you were to _extend_ GCC? Idk, might be laughable